Tallies/TallyMaps/cellMap_class.f90

```diff
@@ -0,0 +1,272 @@
+module cellMap_class
+
+  use numPrecision
+  use genericProcedures,       only : fatalError, numToChar
+  use dictionary_class,        only : dictionary
+  use intMap_class,            only : intMap
+  use particle_class,          only : particleState
+  use outputFile_class,        only : outputFile
+  use tallyMap1D_inter,        only : tallyMap1D, kill_super => kill
+  ! Geometry
+  use geometryStd_class,       only : geometryStd, geometryStd_CptrCast
+  use geometry_inter,          only : geometry
+  use geometryReg_mod,         only : geomPtr
+
+  implicit none
+  private
+
+  !!
+  !! Constructor
+  !!
+  interface cellMap
+    module procedure cellMap_fromDict
+  end interface
+
+  !!
+  !! Map that divides based on the cell a particle is in
+  !! NOTE: the cell ID must be converted from the IDs in the input file
+  !!       to the uniqueIDs.
```

Comment:
Name `uniqueId` is used for the unique number given to each instance of any cell in the geometry. 
For position of the `cell_inter` in the shelf I have been using `cellIdx` 

Basically the number in input file is `cellId`, position in a shelf `cellIdx` and a unique identifier dependent on a position of the isntance in the geometry graph `uniqueId`.